pkg/controllers/dynakube/dtpullsecret/reconciler.go

```diff
@@ -18,27 +21,49 @@ const (
 )
 
 type Reconciler struct {
-	client    client.Client
-	apiReader client.Reader
-	dynakube  *dynatracev1beta2.DynaKube
-	tokens    token.Tokens
+	client       client.Client
+	apiReader    client.Reader
+	dynakube     *dynatracev1beta2.DynaKube
+	tokens       token.Tokens
+	timeprovider *timeprovider.Provider
 }
 
 func NewReconciler(clt client.Client, apiReader client.Reader, dynakube *dynatracev1beta2.DynaKube, tokens token.Tokens) *Reconciler {
 	return &Reconciler{
-		client:    clt,
-		apiReader: apiReader,
-		dynakube:  dynakube,
-		tokens:    tokens,
+		client:       clt,
+		apiReader:    apiReader,
+		dynakube:     dynakube,
+		tokens:       tokens,
+		timeprovider: timeprovider.New(),
 	}
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
-	/* TODO: Introduce cleanup, and also, this would not be needed in every case,
-	if we don't want to do anything with images, then it's not necessary,
-	but other parts of the code must also be updated to handle this scenario.
-	*/
-	if r.dynakube.Spec.CustomPullSecret == "" {
+	if r.dynakube.Spec.CustomPullSecret != "" {
+		return nil
+	}
+
+	if !(r.dynakube.NeedsOneAgent() || r.dynakube.NeedsActiveGate()) {
+		if meta.FindStatusCondition(*r.dynakube.Conditions(), PullSecretConditionType) == nil {
+			return nil // no condition == nothing is there to clean up
+		}
+
+		err := r.deletePullSecretIfExists(ctx)
+		if err != nil {
+			log.Error(err, "failed to clean-up OneAgent tenant-secret")
```

Comment:
i think this log line is not correct 😅 

also if you bubble up the error then don't log it, as it will be logged higher up, and we don't want duplicate error logs

Reply:
We should just log the error and ignore it in the cleanup, as you described in the active gate connection reconciler. 

Will update wrong log line and remove the return so that the conditions are cleanup up as well like in the active gate connection info reconciler.